### ISSUE TITLE
support QAT for PreBN 

### ISSUE THREAD

**Issue by ShiyangZhang** [NVIDIA/TensorRT#3104]
Hi, when resnet is quantized by pytorch_quantization's QAT, it is able to merge bn to conv. However, when BN is before Conv( 'BatchNorm-Conv', used for example in nextvit) , it is unable to merge BN into Conv layer when QDQ node is inserted. Is there any way to achieve this?
```python
class PreBN(nn.Module):
    def __init__(self, num_channels):
        super(PreBN, self).__init__()
        


        self.conv1 = nn.Conv2d(num_channels, num_channels, kernel_size=3, padding=1, bias=False)
        self.bn1 = nn.BatchNorm2d(num_channels)
        self.conv2 = nn.Conv2d(num_channels, num_channels, kernel_size=3, padding=1, bias=False)
        self.bn2 = nn.BatchNorm2d(num_channels)

        self.relu1 = nn.ReLU(inplace=True)
        self.relu2 = nn.ReLU(inplace=True)

        # self.residual_quantizer = quant_nn.TensorQuantizer(quant_nn.QuantConv2d.default_quant_desc_input)

    def forward(self, x):
        out = self.bn1(self.conv1(x))
        
        out = self.relu1((x)+out)
        out = self.relu2(self.conv2(self.bn2(out)))
        return out
```
<img width="324" alt="Screenshot 2023-07-03 at 16 16 16" src="https://github.com/NVIDIA/TensorRT/assets/15400364/f521eb4f-c084-4057-8c67-1599652dd0b8">

![test onnx engine graph json](https://github.com/NVIDIA/TensorRT/assets/15400364/b5386bbe-82c0-4050-b93f-07e7764724d3)

**Comment by zerollzeng** [comment]
@ttyio ^ ^

**Comment by ttyio** [comment]
@ShiyangZhang , we have no `bn-conv` fusion, also the `bn` can commutes with neither `Q` nor `DQ`.  so the best perf here might run the bn in INT8 if we can meet the accuracy requirement. 
The steps might be:
1. modify your quantization code, insert Q/DQ pair also before `bn`.
2. use graph surgeon, remove the `DQ` before `BN` and `Q` after `BN`. The final graph could look like:
            ... -> Relu -> Q -> BN -> DQ -> Conv -> ...

**Comment by ShiyangZhang** [comment]
thx~ By the way, is bn-conv fusion will be supported in the future?

**Comment by ttyio** [comment]
> thx~ By the way, is bn-conv fusion will be supported (without QDQ) in the future?

sorry no plan yet.
